financialaid/views_test.py

```diff
@@ -64,31 +120,130 @@ def test_income_validation_missing_args(self):
         """
         self.data["original_currency"] = "NOTUSD"
-        resp = self.client.post(self.url, self.data)
-        assert resp.status_code == HTTP_400_BAD_REQUEST
+        resp = self.client.post(self.income_validation_url, self.data)
+        assert resp.status_code == status.HTTP_400_BAD_REQUEST
+
+    def test_review_financial_aid_view_not_allowed_user(self):
+        """
+        Tests ReviewFinancialAidView that are not allowed for a user
+        """
+        # Not allowed for default logged-in user
+        resp = self.client.get(self.review_url)
+        assert resp.status_code == status.HTTP_403_FORBIDDEN
+        # Not allowed for staff of different program
+        self.client.force_login(self.staff_user_profile2.user)
+        resp = self.client.get(self.review_url)
+        assert resp.status_code == status.HTTP_403_FORBIDDEN
+        # Not allowed for instructors
+        self.client.force_login(self.instructor_user_profile.user)
+        resp = self.client.get(self.review_url)
+        assert resp.status_code == status.HTTP_403_FORBIDDEN
+        # Not allowed for not-logged-in user
+        self.client.logout()
+        resp = self.client.get(self.review_url)
+        assert resp.status_code == status.HTTP_403_FORBIDDEN
+
+    def test_review_financial_aid_view_not_allowed_program(self):
+        """
+        Tests ReviewFinancialAidView that are not allowed for the program
+        """
+        self.client.force_login(self.staff_user_profile.user)
+        # Not allowed for financial_aid_availability == False
+        self.program.financial_aid_availability = False
+        self.program.save()
+        resp = self.client.get(self.review_url)
+        assert resp.status_code == status.HTTP_404_NOT_FOUND
+        # Not allowed for live == False
+        self.program.financial_aid_availability = True
+        self.program.live = False
+        self.program.save()
+        resp = self.client.get(self.review_url)
+        assert resp.status_code == status.HTTP_404_NOT_FOUND
+        # Reset program
+        self.program.live = True
+        self.program.save()
+        # No valid course_price will return 200, but empty queryset
+        self.course_price.is_valid = False
+        self.course_price.save()
+        resp = self.client.get(self.review_url)
+        assert resp.status_code == status.HTTP_200_OK
+        assert resp.context_data["financial_aid_objects"] == []
+        # Reset course price
+        self.course_price.is_valid = True
+        self.course_price.save()
+
+    def test_review_financial_aid_view_allowed(self):
+        """
+        Tests ReviewFinancialAidView that are allowed
+        """
+        # Allowed for staff of program
+        self.client.force_login(self.staff_user_profile.user)
+        resp = self.client.get(self.review_url)
+        assert resp.status_code == status.HTTP_200_OK
+
+    def test_review_financial_aid_view_with_filter_and_sorting(self):
```

Comment:
Can we test that the sorting and filtering is reflected in the queryset? The queryset output looks to be available in `resp.context_data['object_list']`